ankihub/gui/operations/ankihub_sync.py

```diff
@@ -130,6 +131,24 @@ def _on_sync_done(future: Future, on_done: Callable[[Future], None]) -> None:
         send_review_data, on_done=_on_send_review_data_done
     )
 
+    last_summary_sent_date = config.get_last_summary_sent_date()
+    if not last_summary_sent_date:
+        last_summary_sent_date = (
+            datetime.now().replace(hour=0, minute=0, second=0, microsecond=0)
+            - timedelta(microseconds=1)
+        ) - timedelta(days=1)
+
+    feature_flags = config.get_feature_flags()
+    if (
+        feature_flags.get("daily_card_review_summary", False)
+        and last_summary_sent_date
+        and last_summary_sent_date.date() < datetime.now().date()
+    ):
+        aqt.mw.taskman.run_in_background(
+            lambda: send_daily_review_summaries(last_summary_sent_date),
+            on_done=_on_send_daily_review_summaries_done,
+        )
```

Comment:
:pick: Can we extract this into a `_maybe_send_daily_review_summaries` function?

Reply:
why this function should be private?

Reply:
Because the add-on code base generally follows the convention that if a function is not used outside the file, it should be marked as private/internal by using an underline at the beginning of the function name.

Reply:
but in this case, `send_daily_review_summaries` is being used outside the module `review_data` so in this case the function name should be `maybe_send_daily_review_summaries` instead of `_maybe_send_daily_review_summaries`

Reply:
Anyway, being extremely opinionative here, I don't like the pattern of the "maybe" functions because I consider this harder to understand the function's behavior and hence is harder to debug/maintain. Here is a nice post about it: https://blog.benwinding.com/maybe-functions/. Since you are the developer who spends more time maintaining the addon, I left with you the decision to use the "maybe" function.

Reply:
I think it's fine to use maybe functions in some situations.

> Disadvantages: 
> - Caller has less control, they depend on validation hidden in “maybe function”
> - Hides complexity but doesn’t actually remove it
> - Things that depend on the output now need to handle the maybeness case

> - Things that depend on the output now need to handle the maybeness case

In this case we are not depending on the output of the maybe function (it always returns `None`), so this is not a problem.